### PR TITLE
Longest Increasing subsequence using binary search most optimal approach

### PR DESCRIPTION
**Pull request by namanmodi65**
…ach for this problem

#### Checklist

- [x] Added description of change
- [x] Added file name matches [File name guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#New-File-Name-guidelines)
- [x] Added tests and example, test must pass
- [x] Added documentation so that the program is self-explanatory and educational - [Doxygen guidelines](https://www.doxygen.nl/manual/docblocks.html)
- [x] Relevant documentation/comments is changed or added
- [x] PR title follows semantic [commit guidelines](https://github.com/TheAlgorithms/C-Plus-Plus/blob/master/CONTRIBUTING.md#Commit-Guidelines)
- [x] Search previous suggestions before making a new one, as yours may be a duplicate.
- [x] I acknowledge that all my contributions will be made under the project's license.

Notes:

### CODE REVIEW

File: search/Longest_Increasing_Subsequence_using_binary_search.cpp [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+// Approach 3 :- Using Binary Search
+// Other approaches use additional space to create a new subsequence Array. Instead, this solution uses the existing nums Array to build the subsequence array. We can do this because the length of the subsequence array will never be longer than the current index.
+
+// Time complexity: O(n∗log(n))
+// Space complexity: O(1)
+
+// This approach consider Most optimal Approach for solving this problem
+
+// Implementaion:-
+
```
Comment (realstealthninja) [inline search/Longest_Increasing_Subsequence_using_binary_search.cpp:36]: please document the file using the contributing.md file!

File: search/Longest_Increasing_Subsequence_using_binary_search.cpp [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+#include <iostream>
+#include <vector>
+using namespace std;
```
Comment (realstealthninja) [inline search/Longest_Increasing_Subsequence_using_binary_search.cpp:39]:
```suggestion
```
using namespace std is bad practice and not allowed on this repository!

File: search/Longest_Increasing_Subsequence_using_binary_search.cpp [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            ans.push_back(nums[i]);
+        }
+        else{
+            int idx = lower_bound(ans.begin(),ans.end(),nums[i]) -ans.begin();
+            ans[idx] = nums[i];
+        }
+    }
+    return ans.size();
+}
+
```
Comment (realstealthninja) [inline search/Longest_Increasing_Subsequence_using_binary_search.cpp:57]: please add tests!

**Review by realstealthninja — CHANGES_REQUESTED** [review]

File: search/Longest_Increasing_Subsequence_using_binary_search.cpp (hunk 3 above)
Comment (namanmodi65) [reply search/Longest_Increasing_Subsequence_using_binary_search.cpp:57]: test case are added in main function

Comment (realstealthninja) [reply search/Longest_Increasing_Subsequence_using_binary_search.cpp:57]: move those into a tests function. the main function should be empty except from
return 0 and a call to the tests function

Comment (namanmodi65) [reply search/Longest_Increasing_Subsequence_using_binary_search.cpp:57]: okay

Comment (namanmodi65) [reply search/Longest_Increasing_Subsequence_using_binary_search.cpp:76]: Changes are done please review!

**Pull request closed without merging**